### PR TITLE
feat: hide migrating, retired vaults with no balance

### PR DESCRIPTION
**Pull request by patcito**
hide migrating, retired vaults w/ no balance
 also added a list of hacked vaults to be hidden regardless of balance

### CODE REVIEW

File: app/components/Vault/index.js [hunk 1]
```diff
@@ -763,34 +759,48 @@ const Vault = (props) => {
...
-    apyTooltip = 'Inactive with YIP-56: Buyback and Build';
-  } else if (retired.includes(address)) {
+  // These are our retired v2 vaults, shutting down but not migrating.
+  const retiredv2 = [
+    '0xe11ba472F74869176652C35D30dB89854b5ae84D', // HEGIC v2
+  ];
+
+  const hackedOrToBeAbsolutelyRemoved = [
+    '0xeb231ca94f718daf8ef8e7333b0d690fab9c59d8bc91bc20b5515c77eaff947e', // hacked cySUSD
+  ];
```
Comment (dudesahn) [inline app/components/Vault/index.js:775]: We can remove this since it's not actually a vault, it's for iron bank, so not relevant to v2 website

Comment (patcito) [reply app/components/Vault/index.js]: I removed cySUSD but left the array empty just in case we may need it one day, that ok?

**Review by nymmrx — DISMISSED** [review]

File: app/components/Vault/index.js [hunk 2]
```diff
@@ -763,34 +759,46 @@ const Vault = (props) => {
...
-  // FIXME: migration vaults
-  const retired = [
-    '0x7Ff566E1d69DEfF32a7b244aE7276b9f90e9D0f6', // crvSBTC v1
-    '0x5334e150B938dd2b6bd040D9c4a03Cff0cED3765', // crvRENBTC v1
-    '0xBacB69571323575C6a5A3b4F9EEde1DC7D31FBc1', // crvSAAVE v1
-    '0x7F83935EcFe4729c4Ea592Ab2bC1A32588409797', // crvOBTC v1
-    '0x123964EbE096A920dae00Fb795FFBfA0c9Ff4675', // crvPBTC v1
-    '0xA8B1Cb4ed612ee179BDeA16CCa6Ba596321AE52D', // crvBBTC v1
-    '0x07FB4756f67bD46B748b16119E802F1f880fb2CC', // crvTBTC v1
-    '0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE', // ETH v1
```
Comment (nymmrx) [inline app/components/Vault/index.js:775]: ETH vault is missing from the `migrationWhitelist` json file

File: app/components/Vault/index.js [hunk 3]
```diff
@@ -763,34 +759,46 @@ const Vault = (props) => {
...
+  // Add all vaults here that we only want current holders to see. Include migrating and retiring vaults.
+  let vaultsToHide = migrating.map((v) => v.toLowerCase());
+  vaultsToHide = vaultsToHide
+    .concat(retiredv1.map((v) => v.toLocaleLowerCase()))
+    .concat(retiredv2.map((v) => v.toLocaleLowerCase()));
+  if (migrating.includes(address)) {
     apyRecommended = 'N/A';
-    apyTooltip = 'Please migrate funds to v2 to continue earning yield.';
+    apyTooltip =
+      'Please migrate funds to our new vault to continue earning yield.';
```
Comment (nymmrx) [inline app/components/Vault/index.js:786]: we could use the messages specified in the `migrationWhitelist` json file

File: app/components/Vault/index.js [hunk 4]
```diff
@@ -1268,84 +1276,83 @@ const Vault = (props) => {
...
-      migrationVaultsFrom.includes(vault.address.toLowerCase())) ||
-      !migrationVaultsFrom.includes(vault.address.toLowerCase())) && (
-      <React.Fragment>
-        <Card
-          className={`vault ${amplifyVault ? 'amplify-vault' : ''} ${
-            active ? 'active' : ''
-          } ${vault.isYVBoost ? 'pickle-vault' : ''}`}
-          id={`vault-${accordionKey}`}
+
+  return (!vaultsToHide.includes(vault.address.toLowerCase()) ||
```
Comment (nymmrx) [inline app/components/Vault/index.js:1280]: I think we should filter in a component one level higher as returning `undefined` from a react component is not the best possible practice.

File: app/components/Vault/index.js [hunk 5]
```diff
@@ -763,34 +759,46 @@ const Vault = (props) => {
...
-    // FIXME: yfi vault
-    apyRecommended = 'N/A';
-    apyTooltip = 'Inactive with YIP-56: Buyback and Build';
-  } else if (retired.includes(address)) {
+  // These are our retired v2 vaults, shutting down but not migrating.
+  const retiredv2 = [
+    '0xe11ba472F74869176652C35D30dB89854b5ae84D', // HEGIC v2
+  ];
+
+  const hackedOrToBeAbsolutelyRemoved = [];
```
Comment (nymmrx) [inline app/components/Vault/index.js:773]: Add a one liner documentation, as it was done for `retiredv1` & `retiredv2`

**Review by nymmrx — CHANGES_REQUESTED** [review]
lgtm

File: app/components/Vault/index.js [hunk 6]
```diff
@@ -763,34 +759,46 @@ const Vault = (props) => {
...
+
+  // Vaults that we're migrating to a new version.
+  const migrating = migrationWhitelist.map((v) => v.vaultFrom);
+
+  // Add all vaults here that we only want current holders to see. Include migrating and retiring vaults.
+  let vaultsToHide = migrating.map((v) => v.toLowerCase());
+  vaultsToHide = vaultsToHide
+    .concat(retiredv1.map((v) => v.toLocaleLowerCase()))
+    .concat(retiredv2.map((v) => v.toLocaleLowerCase()));
```
Comment (nymmrx) [inline app/components/Vault/index.js:782]: would be better to work directly with checksummed addresses

File: app/components/Vault/index.js (hunk 4 above)
Comment (nymmrx) [inline app/components/Vault/index.js:1280]: ditto

File: app/components/Vault/index.js [hunk 7]
```diff
@@ -1268,84 +1276,83 @@ const Vault = (props) => {
...
-      <React.Fragment>
-        <Card
-          className={`vault ${amplifyVault ? 'amplify-vault' : ''} ${
-            active ? 'active' : ''
-          } ${vault.isYVBoost ? 'pickle-vault' : ''}`}
-          id={`vault-${accordionKey}`}
+
+  return (!vaultsToHide.includes(vault.address.toLowerCase()) ||
+    vaultBalanceOf > 0) &&
+    !hackedOrToBeAbsolutelyRemoved.includes(vault.address.toLowerCase()) ? (
```
Comment (nymmrx) [inline app/components/Vault/index.js:1282]: ditto

**Review by nymmrx — CHANGES_REQUESTED** [review]

File: app/components/Vault/index.js (hunk 2 above)
Comment (dudesahn) [reply app/components/Vault/index.js:775]: It was never in the migrationWhitelist file before and still migrated just fine; I'm not sure if removing it from the retired list will impact the custom APY tooltip and value though

File: app/components/Vault/index.js (hunk 3 above)
Comment (dudesahn) [reply app/components/Vault/index.js:786]: These messages are here in case they need to be customized

File: app/components/Vault/index.js (hunk 4 above)
Comment (patcito) [reply app/components/Vault/index.js]: It's ok to return null https://reactjs.org/docs/conditional-rendering.html#preventing-component-from-rendering

We could go one level higher but we'd need to move all the balance logic one element higher too which would take time.

File: app/components/Vault/index.js [hunk 8]
```diff
@@ -763,34 +762,52 @@ const Vault = (props) => {
...
-  if (address === '0xBA2E7Fed597fd0E3e70f5130BcDbbFE06bB94fe1') {
-    // FIXME: yfi vault
-    apyRecommended = 'N/A';
-    apyTooltip = 'Inactive with YIP-56: Buyback and Build';
-  } else if (retired.includes(address)) {
+  // These are our retired v1 vaults, shutting down but not migrating.
+  const retiredv1 = retiredv1Json.map((v) => v.address);
+
+  // These are our retired v2 vaults, shutting down but not migrating.
+  const retiredv2 = retiredv2Json.map((v) => v.address);
```
Comment (x48-crypto) [inline app/components/Vault/index.js:769]: Would prefer to use one list for all retired vaults

File: app/components/Vault/index.js [hunk 9]
```diff
@@ -763,34 +761,45 @@ const Vault = (props) => {
...
     apyTooltip = 'This vault was just added or recently updated its strategy.';
-  } else if (address === '0xbD17B1ce622d73bD438b9E658acA5996dc394b0d') {
+  } else if (vaultIsPickle) {
     apyRecommended = 'N/A';
     apyTooltip =
       'Please migrate funds to yvBOOST-ETH to continue earning maximum yield.';
+  } else if (retired.includes(address)) {
+    apyRecommended = 'N/A';
+    apyTooltip = retiredTooltips[address];
+  } else if (retired.includes(address)) {
```
Comment (x48-crypto) [inline app/components/Vault/index.js:800]: Remove duplicate conditional logic

**Review by x48-crypto — APPROVED** [review]

**Review by nymmrx — APPROVED** [review]
lgtm

**Pull request merged**